axelor-account/src/main/java/com/axelor/apps/account/service/debtrecovery/DoubtfulCustomerService.java

```diff
@@ -198,24 +209,145 @@ public void createDoubtFulCustomerMove(
+        copyOnNewMoveLine.setAmount(remainingAmount);
+        BigDecimal newPercentage =
+            remainingAmount
+                .multiply(percentage)
+                .divide(amount, AppBaseService.DEFAULT_NB_DECIMAL_DIGITS, RoundingMode.HALF_UP);
+        copyOnNewMoveLine.setPercentage(newPercentage);
+        invoiceTermToAddToDebitMoveLine.add(copyOnNewMoveLine);
+        invoiceTermToAdd.add(copyOnNewMoveLine);
+
+        InvoiceTerm copyOnOldMoveLine = invoiceTermRepo.copy(invoiceTerm, false);
+        copyOnOldMoveLine.setAmount(remainingAmount);
+        copyOnOldMoveLine.setAmountRemaining(remainingAmount);
+        copyOnOldMoveLine.setPercentage(newPercentage);
+        invoiceTermToAddToInvoicePartnerMoveLine.add(copyOnOldMoveLine);
+        invoice.addInvoiceTermListItem(copyOnOldMoveLine);
+        invoiceTermToRemove.add(copyOnOldMoveLine);
+        // Update current invoice term
+        invoiceTerm.setAmount(amount.subtract(remainingAmount));
+        invoiceTerm.setAmountRemaining(BigDecimal.ZERO);
+        invoiceTerm.setIsPaid(true);
+        invoiceTerm.setPercentage(percentage.subtract(newPercentage));
+      }
+    }
+
+    for (InvoiceTerm it : invoiceTermToAddToInvoicePartnerMoveLine) {
+      invoicePartnerMoveLine.addInvoiceTermListItem(it);
+    }
 
     if (creditMoveLine != null) {
+      // Create invoice term on new credit move line
+      this.createInvoiceTerm(
+          creditMoveLine, bankDetails, pfpUser, paymentMode, todayDate, amountRemaining);
+
+      newMove.getMoveLineList().add(debitMoveLine);
+      newMove.getMoveLineList().add(creditMoveLine);
+
+      moveValidateService.accounting(newMove);
+      moveRepo.save(newMove);
+
       Reconcile reconcile =
           reconcileService.createReconcile(
               invoicePartnerMoveLine, creditMoveLine, amountRemaining, false);
+
       if (reconcile != null) {
         reconcileService.confirmReconcile(reconcile, true);
       }
     }
+    if (debitMoveLine.getInvoiceTermList() != null) {
+      for (InvoiceTerm it : debitMoveLine.getInvoiceTermList()) {
+        invoiceTermRepo.remove(it);
+      }
+      debitMoveLine.clearInvoiceTermList();
+    }
+    for (InvoiceTerm it : invoiceTermToAddToDebitMoveLine) {
+      debitMoveLine.addInvoiceTermListItem(it);
+    }
+    // Recalculate percentage on debit moveline
+    for (InvoiceTerm it : debitMoveLine.getInvoiceTermList()) {
+      it.setPercentage(
+          it.getAmount()
+              .multiply(BigDecimal.valueOf(100))
+              .divide(
+                  amountRemaining, AppBaseService.DEFAULT_NB_DECIMAL_DIGITS, RoundingMode.HALF_UP));
+    }
+
+    if (invoice != null) {
+      for (InvoiceTerm it : invoiceTermToAdd) {
+        invoice.addInvoiceTermListItem(it);
+      }
+
+      for (InvoiceTerm it : invoiceTermToRemove) {
+        invoice.removeInvoiceTermListItem(it);
+        it.setInvoice(null);
+      }
+    }
```

Comment:
This code needs to be moved to another service (InvoiceTermService maybe or a new service for invoice term for doubtful customers feature)